lib/services/livesync/livesync-service-base.ts

```diff
@@ -212,9 +217,12 @@ class LiveSyncServiceBase implements ILiveSyncServiceBase {
 					}
 
 					this.$logger.info("Applying changes...");
-					platformLiveSyncService.refreshApplication(deviceAppData, localToDevicePaths, data.forceExecuteFullSync || !wasInstalled).wait();
+					if (!this.emit("sync", device, data)) {
```

Comment:
According to nodejs documentation:
```
Returns true if the event had listeners, false otherwise.
```
https://nodejs.org/api/events.html#events_emitter_emit_eventname_arg1_arg2

So the refreshApplication method will be called only if there are no listeners (currently every other case except debug command I think). Is this the expected behavior?